src/ext/dispatch_compat_php5.c

```diff
@@ -328,23 +329,39 @@ int ddtrace_forward_call(zend_execute_data *execute_data, zend_function *fbc, zv
 }
 
 void ddtrace_copy_function_args(zend_execute_data *execute_data, zval *user_args) {
-    INIT_ZVAL(*user_args);
-    // @see https://github.com/php/php-src/blob/PHP-5.4/Zend/zend_builtin_functions.c#L447-L473
+    /* This is taken from func_get_args
+     * PHP 5.3 - 5.5 are the same:
+     * @see https://github.com/php/php-src/blob/PHP-5.4/Zend/zend_builtin_functions.c#L445-L473
+     * In 5.6 it changed:
+     * @see https://github.com/php/php-src/blob/PHP-5.6/Zend/zend_builtin_functions.c#L443-L476
+     */
     void **p = EX(function_state).arguments;
     if (p && *p) {
         int arg_count = (int)(zend_uintptr_t)*p;
         array_init_size(user_args, arg_count);
         for (int i = 0; i < arg_count; i++) {
+#if PHP_VERSION_ID < 50600
             zval *element;
 
             ALLOC_ZVAL(element);
             *element = **((zval **)(p - (arg_count - i)));
             zval_copy_ctor(element);
             INIT_PZVAL(element);
+#else
+            zval *element, *arg;
+
+            arg = *((zval **)(p - (arg_count - i)));
+            if (!Z_ISREF_P(arg)) {
+                element = arg;
+                Z_ADDREF_P(element);
+            } else {
+                ALLOC_ZVAL(element);
+                INIT_PZVAL_COPY(element, arg);
+                zval_copy_ctor(element);
+            }
+#endif
             zend_hash_next_index_insert(Z_ARRVAL_P(user_args), &element, sizeof(zval *), NULL);
         }
-    } else {
-        array_init_size(user_args, 0);
```

Comment:
If the args is typed as `array`, then passing null might be an issue? Did you test this? I would expect it to fail as `null` is not compatible with `array`.

Reply:
I ran it on PHP 5.6 and PHP 7.3 builds of this branch. I'm not sure how it will function on PHP 7 if we return NULL - especially after nullables were introduced. But what are your thoughts on the different behaviors?

Reply:
I think I'll just put the empty array back. It seems odd that the parameter says it expects `array` but null gets passed, and it doesn't fail.